server/src/main/java/org/apache/iotdb/db/query/executor/fill/PreviousFill.java

```diff
@@ -77,6 +80,18 @@ void constructFilter() {
     timeFilter = FilterFactory.and(lowerBound, TimeFilter.ltEq(queryTime));
   }
 
+  public void convertRange(long startTime) {
+    if (beforeRange % GroupByEngineDataSet.MS_TO_MONTH == 0) {
+      Calendar calendar = Calendar.getInstance();
```

Comment:
Please use the new date and time API defined in java8. See org.apache.iotdb.db.qp.utils.DatetimeUtils for examples.